ProfileManager.js

```diff
@@ -107,10 +117,11 @@ export default class ProfileManager {
       for(const zcap of Object.values(profileAgent.zcaps)) {
         content.zcaps[zcap.referenceId] = zcap;
       }
-      return content;
+      agent = content;
     }
 
-    return await this._getAgentContent({profileAgentRecord});
+    this._cache.agents.set(profileId, agent);
```

Comment:
it would be good if there is anything that can be done to make it more clear that profile agents are being indexed by `profileId`.  My intuition with this sort of namespacing would be that the index would be `profileAgentId`.

Reply:
So something like `_cache.agentsByProfileId`?